Source/Core/Core/PowerPC/PPCSymbolDB.cpp

```diff
@@ -399,6 +402,26 @@ bool PPCSymbolDB::LoadMap(const Core::CPUThreadGuard& guard, const std::string&
     if (name[strlen(name) - 1] == '\r')
       name[strlen(name) - 1] = 0;
 
+    // Split the current name string into separate parts, and get the object name
+    // if it exists.
+    std::vector<std::string> parts = SplitString(name, '\t');
+    size_t num_parts = parts.size();
+
+    std::string object_filename_string = "";
+    std::string name_string = std::string(StripWhitespace(parts[0]));
+
+    if (num_parts > 1)
+    {
+      std::vector<std::string> info_parts = SplitString(parts[1], ' ');
+      size_t num_info_parts = info_parts.size();
+
+      // If the last part contains a ., it has to be the object name
+      if (info_parts[num_info_parts - 1].find('.') != std::string::npos)
+      {
+        object_filename_string = info_parts[num_info_parts - 1];
+      }
+    }
```

Comment:
This doesn't take into account symbols with multiple objects nor the ones without `.` _(e.g. linker generated symbols)_.

Won't the following be enough or have you come across symbol map files where it doesn't work?
```C++
const std::vector<std::string> parts = SplitString(name, '\t');
const std::string name_string{StripWhitespace(parts[0])};
const std::string object_string{(parts.size() > 1) ? StripWhitespace(parts[1]) : ""};
```

Reply:
This includes the .a library filename as well, which is not what I want here; I only want to get the object filename. (e.g. "library.a file.o" -> file.o).

Reply:
I decided to just lump in the library filename at least for now.